File: V семестр/Паралельне програмування/Кочетов/лабы/lab3/lab3/F3.cs
using System;

namespace lab3
{
    public class F3
    {
    private Vector result;
    public Vector getResult()
    {
        return result;
    }

    private int N;

    public F3(int N)
    {
        this.N = N;
    }

    public void run()
    {
            Console.WriteLine("Task 3 start");
            Vector S = new Vector(N);
            Matrix MO = new Matrix(N), MP = new Matrix(N), MR = new Matrix(N);
            result = MO.multiply(MP).multiply(S).sum(MR.multiply(S.sort()));
            Console.WriteLine("Task 3 end");
    }
}

}

File: V семестр/Паралельне програмування/Кочетов/лабы/lab6/lab6/F3.cpp
#include "F3.h"

F3::F3(int N) {
	this->N = N;
}

Vector* F3::getResult() {
	return result;
}

void F3::run() {
	cout << "Task 3 start\n";
	Vector* S = new Vector(N);
	Matrix *MO = new Matrix(N), *MP = new Matrix(N), *MR = new Matrix(N);
	result = MO->multiply(MP)->multiply(S)->sum(MR->multiply(S->sort()));
	cout << "Task 3 end\n";
	delete S;
	delete MO;
	delete MP;
	delete MR;
}